src/main/java/hw4/pages/MetalsColorsPage.java

```diff
@@ -0,0 +1,167 @@
+
+    @FindBy(id = "submit-button")
+    private WebElement submitButton;
+
+
+    public MetalsColorsPage(WebDriver driver) {
+        super(driver);
+        result = new Result(driver);
+        PageFactory.initElements(driver, this);
+    }
+
+    public void fillForm(FormData form) {
+        setColor(form.getColors());
+        setMetal(form.getMetals());
+        setVegetable(form.getVegetables());
+        setElement(form.getElements());
+        setEven(form.getEven());
+        setOdd(form.getOdd());
+    }
+
+    public void setColor(String color) {
+        if (!color.equalsIgnoreCase(Colors.COLORS.toString())) {
+            setWebElement(colors, colorsDropdownElements, colors, color);
+        }
+    }
+
+    public void setMetal(String metal) {
+        if (!metal.equalsIgnoreCase(Metals.METALS.toString())) {
+            setWebElement(metalsCaret, metalsDropdownElements, metals, metal);
+        }
+    }
+
+    public void setVegetable(List<Vegetables> vegetable) {
+        if (!vegetable.get(0).toString().equalsIgnoreCase(Vegetables.VEGETABLES.toString())) {
+            wait.until(ExpectedConditions.elementToBeClickable(vegetables)).click();
+            wait.until(ExpectedConditions
+                    .attributeToBe(By.cssSelector("#vegetables .dropdown-menu"), "style", "display: block;"));
+            // to uncheck vegetables checkbox, because by default it is checked
+            driver.findElement(By.xpath("//*[@id='vegetables']//label[text()='Vegetables']")).click();
+            for (Vegetables vegetableToSet : vegetable) {
+                for (WebElement vegetablesDropdownElement : vegetablesDropdownElements) {
+                    if (vegetablesDropdownElement.getText().equalsIgnoreCase(vegetableToSet.toString())) {
+                        vegetablesDropdownElement.click();
+                    }
+                }
+            }
+        }
+    }
+
+    public void setElement(List<Elements> element) {
+        if (element != null) {
+            wait.until(ExpectedConditions.visibilityOfAllElements(elements));
+            for (Elements elemToSet : element) {
+                for (WebElement webElement : elements) {
+                    if (webElement.getText().equalsIgnoreCase(elemToSet.toString())) {
+                        webElement.click();
+                    }
+                }
+            }
+        }
+    }
+
+    public void setOdd(String value) {
+        setNumber(value, summaryRadioOdd);
+    }
+
+    public void setEven(String value) {
+        setNumber(value, summaryRadioEven);
+    }
+
+    private void setNumber(String value, List<WebElement> whereToSet) {
+        if (!value.equals("2") | !value.equals("1")) {
+            wait.until(ExpectedConditions.visibilityOfAllElements(whereToSet));
+            for (WebElement webElement : whereToSet) {
+                if (webElement.getText().equalsIgnoreCase(value)) {
+                    webElement.click();
+                }
```

Comment:
I see that you have got a lot of places where you use this loop. Looks like code duplication

Reply:
Thanks, I refactored my code.